ruby/example_code/s3/auth_federation_token_request_test.rb

```diff
@@ -1,142 +1,88 @@
- #* CONDITIONS OF ANY KIND, either express or implied. See the License for the
- #* specific language governing permissions and limitations under the License.
-#**
-# snippet-sourcedescription:[auth_federation_token_request_test.rb allows a federated user with a limited set of permissions to lists keys in the specified bucket.] 
-# snippet-service:[s3]
-# snippet-keyword:[Ruby]
-# snippet-sourcesyntax:[ruby]
-# snippet-keyword:[Amazon S3]
-# snippet-keyword:[Code Sample]
-# snippet-keyword:[GET Bucket, GET Object, GET Keys]
-# snippet-sourcetype:[full-example]
-# snippet-sourcedate:[2019-02-11]
-# snippet-sourceauthor:[AWS]
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX - License - Identifier: Apache - 2.0
+
+# This code example allows a federated user with a limited set of
+# permissions to list the objects in the specified Amazon S3 bucket.
+
 # snippet-start:[s3.ruby.auth_federation_token_request_test.rb]
 require 'aws-sdk-s3'
 require 'aws-sdk-iam'
 
-USAGE = <<DOC
-
-Usage: ruby auth_federation_token_request_test.rb -b BUCKET -u USER [-r REGION] [-d] [-h]
-
-  Creates a federated policy for USER to list items in BUCKET for one hour.
-
-  BUCKET is required and must already exist.
-
-  USER is required and if not found, is created.
-
-  If REGION is not supplied, defaults to us-west-2.
-
-  -d gives you extra (debugging) information.
-
-  -h displays this message and quits.
-
-DOC
-
-def print_debug(debug, s)
-  if debug
-    puts s
-  end
-end
-
-# Get the user if they exist, otherwise create them
-def get_user(region, user_name, debug)
-  iam = Aws::IAM::Client.new(region: 'us-west-2')
-  
-  # See if user exists
-  user = iam.user(user_name)
- 
-  # If user does not exist, create them
-  if user == nil
-    user = iam.create_user(user_name: user_name)
-    iam.wait_until(:user_exists, user_name: user_name)
-    print_debug(debug, "Created new user #{user_name}")
-  else
-    print_debug(debug, "Found user #{user_name} in region #{region}")
+# Uses an AWS IAM user with a limited set of persmissions to list the objects
+# in the specified Amazon S3 bucket.
+# @param region [String] The AWS Region to use for configuring the AWS STS
+#   and S3 clients.
+# @param user_name [String] The user's name.
+# @param bucket_name [String] The bucket's name.
+# @return [Boolean] true if all operations succeed; otherwise, false.
+# @example
+#   list_objects_with_temporary_credentials('us-east-1', 'my-user', 'my-bucket')
+def list_objects_with_temporary_credentials(region, user_name, bucket_name)
+  # Create a new AWS STS client and get temporary credentials.
+  outcome = false
+  sts = Aws::STS::Client.new(region: region)
+  creds = ''
+
+  begin
+    creds = sts.get_federation_token({
+      duration_seconds: 3600,
+      name: user_name,
+      policy: "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Sid\":\"Stmt1\",\"Effect\":\"Allow\",\"Action\":\"s3:ListBucket\",\"Resource\":\"arn:aws:s3:::#{bucket_name}\"}]}"
```

Comment:
for readability you may want to define these in the future as a multiline Ruby hash then use `to_json` to convert it to a json string. you would have to require the json library at the top, which is normal.

also with Rubocop, 80 character lines are preferred.

```
irb(main):004:0> require 'json'
=> true
irb(main):005:0> { Version: '1-2-3', Key: 'Value' }.to_json
=> "{\"Version\":\"1-2-3\",\"Key\":\"Value\"}"
```

Reply:
I converted this to a Ruby hash.